packages/expo-device/ios/EXDevice/EXDevice.m

```diff
@@ -0,0 +1,252 @@
+// Copyright 2018-present 650 Industries. All rights reserved.
+
+#import <EXDevice/EXDevice.h>
+#import "DeviceUID.h"
+#import <UMCore/UMUtilities.h>
+
+#include <ifaddrs.h>
+#include <arpa/inet.h>
+#import <mach-o/arch.h>
+#import <CoreLocation/CoreLocation.h>
+#import <UIKit/UIKit.h>
+#import <sys/utsname.h>
+
+#import <CoreTelephony/CTCallCenter.h>
+#import <CoreTelephony/CTCall.h>
+#import <CoreTelephony/CTCarrier.h>
+#import <CoreTelephony/CTTelephonyNetworkInfo.h>
+
+#import <UMCore/UMUtilitiesInterface.h>
+#import <UMCore/UMUtilities.h>
+
+#import <WebKit/WKWebView.h>
+
+#if !(TARGET_OS_TV)
+#import <LocalAuthentication/LocalAuthentication.h>
+#endif
+
+@interface EXDevice() {
+  WKWebView *webView;
+}
+
+@property (nonatomic, strong) NSString *webViewUserAgent;
+
+@end
+
+@implementation EXDevice
+
+UM_EXPORT_MODULE(ExpoDevice);
+
+- (dispatch_queue_t)methodQueue
+{
+  return dispatch_get_main_queue();
+}
+
+- (void)setModuleRegistry:(UMModuleRegistry *)moduleRegistry
+{
+  _moduleRegistry = moduleRegistry;
+}
+
+UM_EXPORT_METHOD_AS(getUserAgentAsync,
+                    getWebViewUserAgentWithResolver:(UMPromiseResolveBlock)resolve
+                    rejecter:(UMPromiseRejectBlock)reject)
+{
+  __weak EXDevice *weakSelf = self;
+  dispatch_async(dispatch_get_main_queue(), ^{
+    __strong EXDevice *strongSelf = weakSelf;
+    if (strongSelf) {
+      if (!strongSelf.webViewUserAgent) {
```

Comment:
I think you could coalesce these two ifs into one, `(nil).webViewUserAgent` should return `nil` without failing, also would make the promise resolve/reject even if `strongSelf` is `nil`

Reply:
@EvanBacon were there any edge cases caught by using two `if` statements?